src/D2L.CodeStyle.Analyzers/Immutability/StatelessFuncAnalyzer.cs

```diff
@@ -165,19 +225,45 @@ ExpressionSyntax expression
 			ISymbol symbol,
 			ImmutableHashSet<ISymbol> statelessFuncs
 		) {
-
-			if( statelessFuncs.Contains( symbol ) ) {
```

Comment:
Was this just an unnecessary check?

Reply:
Ah, we're calling this on arguments now, not constructors. So the type info is different, and this became superfluous.

Reply:
I think it was also always unnecessary yes. The OriginalDefinition of the original defintion, is still itself, so it's always okay to just do OriginalDefinition.